lib/codebreaker/game_process.rb

```diff
@@ -0,0 +1,157 @@
+require "codebreaker/game"
+require "yaml"
+
+module Codebreaker
+  CODE_REGEXP = /^[1-6]{4}$/
+  DB_FILE = "scores.yaml"
+  class GameProcess
+    
+    STAGES_CHANGE = {
+      menu: {
+        "new game" => :game,
+        "scores" => :scores,
+        "exit" => :exit
+      }, 
+      game: {
+        "win" => :complete_game,
+        "lose" => :complete_game,
+        "restart" => :game
+      },      
+      complete_game: {
+        "yes" => :save_score,
+        "no" => :repeate
+      },
+      save_score: Hash.new(:repeate),
+      repeate: {
+        "yes" => :game,
+        "no" => :menu
+      },
+      scores: {
+        "back" => :menu
+      }
+    }
+
+    attr_reader :stage, :answers
+
+    def initialize
+      @game = Game.new
+      @stage = :menu
+      @stage_changes = false
+    end     
+
+    def remaining_guess
+      @game.guesses_quantity if @stage == :game
```

Comment:
@stage == :game is using in more than one place so we can move it two meaningful named method